product_price_category/models/product_pricelist_item.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2017 Camptocamp SA
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import api, fields, models
+
+
+class ProductPricelistItem(models.Model):
+    _inherit = "product.pricelist.item"
+
+    applied_on = fields.Selection(
+        selection_add=[("2b_product_price_category", "Price Category")],
+        ondelete={"2b_product_price_category": "set default"},
+    )
+    price_category_id = fields.Many2one(
+        comodel_name="product.price.category",
+        string="Price Category",
+        ondelete="cascade",
+        help="Specify a product price category if this rule only applies "
+        "to one price category. Keep empty otherwise.",
+    )
+
+    @api.onchange("applied_on")
+    def _onchange_applied_on_price_category(self):
+        """Reset the price_category_id value if applied_on
+        is not price_category
+        """
+        if self.applied_on != "2b_product_price_category":
+            self.price_category_id = False
```

Comment:
Shouldn't be the 'price_category_id' field converted into computed field readonly False in order to remove onchange?